src/Engines/ElasticSearchEngine.php

```diff
@@ -197,7 +197,7 @@ private function performSearch(BaseBuilder $builder, $options = [])
                 $callback,
                 $this->elasticsearch,
                 $searchBody
-            );
+            )->asArray();
```

Comment:
I think same approach before but how could we be sure this callback return an object and it has asArray method?

Reply:
So does that mean that omitting `asArray` call was intentional? As in it is left up to the callback to decide what is returned?

Reply:
And actually now that I think, its a callback. There's literally no way to make sure that it will return what you expect, cause ultimately the developer really can return whatever they want, and I am not aware of any way in php to enforce a return value of callback.

So I think maybe the only thing i can do, is add the appropriate type hint via annotation there?

Reply:
Yes, while implementing upgrade I faced same problem and could not find a proper solution and I keep it as is without asArray(). What it can be, we may use invoke method of a class. The invoke could have return type.